src/server/channel_store.cc

```diff
@@ -152,7 +153,7 @@ unsigned ChannelStore::SendMessages(std::string_view channel, facade::ArgRange m
       it++;
     }
   };
-  shard_set->pool()->DispatchBrief(std::move(cb));
+  shard_set->pool()->AwaitBrief(std::move(cb));
```

Comment:
This funciton is called inside DbSlice::DeleteExpiredStep which we asume do not preempt but now this breaks this assumption

Reply:
:thinking: Then we can:

1. Use `DispatchBrief` only when called by `DeleteExpiredStep`
2. Ditch strong consistency when sending messages via publish and add a sleep to the test

I rather opt in for (1).

Other than that good catch! 

Wdyt ? @adiholden

Reply:
I think we should add sleep to the test case as from what I understand the publish does not guaranties to send the message to all subscribers but that the message was queued to be sent.
Also when sending the message we use the conn->SendPubMessageAsync which I think also does not guarantee that after you call AwaitBrief here the massege was already sent

Reply:
Sounds good